pkg/helm/release/manager.go

```diff
@@ -226,15 +226,22 @@ func installRelease(ctx context.Context, tiller *tiller.ReleaseServer, namespace
 		// Workaround for helm/helm#3338
 		if releaseResponse.GetRelease() != nil {
 			uninstallReq := &services.UninstallReleaseRequest{
-				Name:  releaseResponse.GetRelease().GetName(),
+				Name:  name,
 				Purge: true,
 			}
 			_, uninstallErr := tiller.UninstallRelease(ctx, uninstallReq)
-			if uninstallErr != nil {
-				return nil, fmt.Errorf("failed to roll back failed installation: %s: %s", uninstallErr, err)
+
+			// In certain cases, InstallRelease will return a partial release in
+			// the response even when it doesn't record the release in its release
+			// store. In that case the rollback will fail with a not found error
+			// because there was nothing to rollback. So we check for that and
+			// skip logging the rollback error when the error was that it didn't
+			// exist.
+			if uninstallErr != nil && !notFoundErr(uninstallErr) {
+				return nil, fmt.Errorf("failed installation (%s) and failed rollback (%s)", err, uninstallErr)
 			}
 		}
-		return nil, err
+		return nil, fmt.Errorf("failed to install release: %s", err)
```

Comment:
Can we include the release name here in the eror as well as we do have it a bit above?

Reply:
I could. Only reason I didn't is because the [release name is already included in the structured log](https://github.com/operator-framework/operator-sdk/blob/45e5ca3a4eeca9067f966b9d52d7bbff2298e40e/pkg/helm/controller/reconcile.go#L86) as a separate field when we log this error here:

https://github.com/operator-framework/operator-sdk/blob/45e5ca3a4eeca9067f966b9d52d7bbff2298e40e/pkg/helm/controller/reconcile.go#L172

In fact, I'm now seeing that we already log "Failed to install release" so I think I'll revert this and the `failed to update` changes. WDYT?

Reply:
Alright fixed. I realized the "install failed" and "update failed" pieces of the message were only ever being logged in our log message, but not in the controller log that happens when an error is returned from the `Reconcile` function.

So I opted to simplify our log and leave the details in the error message. WDYT?